bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -462,17 +463,12 @@ def neuron_stats_update(self, neuron_stats: Dict[int, Dict[str, Any]]):
                 stats.setdefault('updates!', 1)  # number of EMA zeroing updates init to zero
 
             for key in self.synapse_keys:
-                zkey = key + '!'
-                if zkey in stats:
-                    if key in _stats:
-                        stats[zkey] = (1 - self.alpha) * stats[zkey] + self.alpha * _stats[key]
-                    else:
-                        stats[zkey] = (1 - self.alpha) * stats[zkey]  # + self.alpha * 0
+                zkey = key + '!'  # zeroing key
```

Comment:
What is this zkey? Is that how we are indexing the stats?

Reply:
The zeroing key is a copy of a stat where unsuccessful responses mean a zero will EMA into the value, thus decaying the neuron's key statistic according to which it is weighted. The original key statistic does not push a zero on unsuccessful responses, but only on valid responses, such that the original key is an accurate representation of neuron performance when responsive.